spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -175,7 +184,37 @@ public Statistics estimateStatistics() {
   protected Statistics estimateStatistics(Snapshot snapshot) {
     // its a fresh table, no data
     if (snapshot == null) {
-      return new Stats(0L, 0L);
+      return new Stats(0L, 0L, Collections.emptyMap());
+    }
+
+    boolean cboEnabled =
+        Boolean.parseBoolean(spark.conf().get(SQLConf.CBO_ENABLED().key(), "false"));
+    Map<NamedReference, ColumnStatistics> colStatsMap = null;
+    if (readConf.enableColumnStats() && cboEnabled) {
+      colStatsMap = Maps.newHashMap();
+      List<StatisticsFile> files = table.statisticsFiles();
+      if (!files.isEmpty()) {
+        List<BlobMetadata> metadataList = (files.get(0)).blobMetadata();
+
+        for (BlobMetadata blobMetadata : metadataList) {
+          int id = blobMetadata.fields().get(0);
+          String colName = table.schema().findColumnName(id);
+          NamedReference ref = FieldReference.column(colName);
+
+          long ndv = 0;
+          String ndvStr = blobMetadata.properties().get("ndv");
+          if (ndvStr != null && !ndvStr.isEmpty()) {
+            ndv = Long.parseLong(blobMetadata.properties().get("ndv"));
+          } else {
+            LOG.debug("ndv is not set in BlobMetadata for column {}", colName);
+          }
+
+          // TODO: Fill min, max and null from the manifest file
+          ColumnStatistics colStats = new SparkColumnStatistics(ndv, null, null, 0L, 0L, 0L, null);
```

Comment:
When ndv is not set, we are sending in 0. Is that intended?

Reply:
Yeah I didn't see notes in the Spark docs about what 0 would mean in this context.

Reply:
Checked Spark code, should be `None` if ndv is not available. I changed the code accordingly.

Reply:
Looks like it is still 0?

Reply:
Changed to `null`.